compiler/rustc_codegen_llvm/src/debuginfo/mod.rs

```diff
@@ -103,14 +103,14 @@ impl<'ll, 'tcx> CodegenUnitDebugContext<'ll, 'tcx> {
             // for macOS to understand. For more info see #11352
             // This can be overridden using --llvm-opts -dwarf-version,N.
             // Android has the same issue (#22398)
-            if let Some(version) = sess.target.dwarf_version {
-                llvm::LLVMRustAddModuleFlag(
-                    self.llmod,
-                    llvm::LLVMModFlagBehavior::Warning,
-                    "Dwarf Version\0".as_ptr().cast(),
-                    version,
-                )
-            }
+            let dwarf_version =
+                sess.opts.debugging_opts.dwarf_version.unwrap_or(sess.target.default_dwarf_version);
+            llvm::LLVMRustAddModuleFlag(
+                self.llmod,
+                llvm::LLVMModFlagBehavior::Warning,
+                "Dwarf Version\0".as_ptr().cast(),
+                dwarf_version,
+            );
```

Comment:
Could this have caused #99143? Looks like before this PR, we only ever set `Dwarf Version` on targets with `Some` in their `dwarf_version`, which based on the diff in this PR, wasn't the case for e.g. Windows (at least, there's probably more targets not covered by the `${OS}_base.rs` files).

Reply:
Alright that was confusing but I'm pretty sure (https://github.com/rust-lang/rust/issues/99143#issuecomment-1182315377) that what's going on is even with debuginfo disabled for a binary, the standard library is getting statically linked in *and we ship it w/ debuginfo*.

I'm not sure what the best approach here would be, maybe just relying on the `is_like_msvc` check to effectively switch between `CodeView` and `Dwarf Version` (and never set both)?